packages/calcite-components/src/utils/logger.ts

```diff
@@ -0,0 +1,84 @@
+import { logLevel } from "./config";
+
+export type LogLevel = "debug" | "info" | "warn" | "error" | "trace";
+
+type Message = string;
+type MajorVersion = number;
+
+type DeprecatedContext = "component" | "property" | "method" | "event" | "slot";
+type DeprecatedParams = {
+  name: string;
+  suggested?: string | string[];
+  component?: string;
+  removalVersion: MajorVersion | "future";
+};
+
+/**
+ * Exported for testing purposes only
+ */
+export const loggedDeprecations = new Set<string>();
+
+const logLevels = {
+  debug: 1,
+  info: 2,
+  warn: 4,
+  error: 8,
+} as const;
+
+function willLog(level: LogLevel): boolean {
+  return logLevels[level] >= logLevels[logLevel];
+}
+
+function forwardToConsole(level: LogLevel, ...data: any[]): void {
+  if (!willLog(level)) {
+    return;
+  }
+
+  const badgeTemplate = "%ccalcite";
+  const badgeStyle = "background: #007AC2; color: #fff; border-radius: 4px;padding: 2px 4px;";
```

Comment:
Should design approve the styling :)

Wondering if we could use CSS vars here instead of styles. I guess it doesn't really matter since its just a log badge but still.

Reply:
I did test this, but no dice. Custom CSS props don't appear to be available for console styling. 😞